other/web-js/bluehawk.sh

```diff
@@ -2,10 +2,135 @@
 
 pushd "$(dirname "$0")"
 
-rm -r generated
+states=("prod-graphql" "prod-mql" "prod-data-api")
+root_dir=$PWD
 
-for state in prod-mql prod-graphql; do
+function rename_project() {
+  local project_name=$1
+  jq ".name = \"$project_name\"" package.json >> package.json.tmp
+  mv package.json.tmp package.json
+}
+
+function remove_unused_files() {
+  local unused_files=("$@")
+  for filepath in "${unused_files[@]}"; do
+    rm $filepath
+  done
+}
+
+function remove_unused_dependencies() {
+  local unused_dependencies=("$@")
+  # This uses jq to update package.json. We don't use `npm uninstall`
+  # because that also installs all the other packages, which takes much
+  # longer and leaves you with a new node_modules folder to clean up.
+
+  # Construct a jq expression to remove the unused dependencies. The
+  # final expression resemables the following:
+  # jq_expr="del(.dependencies[\"bson\"], .dependencies[\"crypto-browserify\"], .dependencies[\"jwt-decode\"])"
+  local jq_expr=""
+  for dependency in "${unused_dependencies[@]}"; do
+    dep_str=".dependencies[\"$dependency\"]"
+    if [ -z "$jq_expr" ]; then
+      jq_expr=$dep_str
+    else
+      jq_expr="${jq_expr}, ${dep_str}"
+    fi
+  done
+  jq_expr="del(${jq_expr})"
+
+  jq "$jq_expr" package.json >> package.json.tmp
+  mv package.json.tmp package.json
+}
+
+function modify_craco_config() {
```

Comment:
we should migrate these templates from CRA to another web framework sometime soonish b/c CRA is no longer actively maintained...not for this PR tho